Update README.md with link to trestle-omniauth

airhorns opened the pull request:
I authored and have been using `trestle-omniauth` for authenticating into various Trestle admins using various single sign on providers, like `omniauth-google-oauth2` and whatnot. If you already have GSuite or something set up for your admin users, it's handy to not have to maintain another account! `trestle-omniauth` is implemented similarly to `trestle-auth` and adds a `config.omniauth` to Trestle's config allowing adding omniauth providers.

I think others might want to do the same so I figured it'd be good to make it a bit more discoverable by adding it to the main Trestle readme. `trestle-omniauth` is pretty fresh however, and I think I'm really the only user, so if it's too early for this tacit endorsement I totally understand!

Thanks for a productive piece of software!

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TrestleAdmin/trestle?pullRequest=281) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/TrestleAdmin/trestle?pullRequest=281) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TrestleAdmin/trestle?pullRequest=281) it.</sub>

spohlenz commented:
I've had a quick review, but unfortunately wasn't able to get it working (just testing on a new app with the `developer` provider and no existing OmniAuth integration).

Once I can see it in action I'd be happy to add it to the README but did have a few questions/comments:

1) How are you configuring OmniAuth to use the `sessions#create` action (which is mounted by default at `/admin/auth/:provider/callback`)?

2) How are you authorizing that an OmniAuth-authenticated user is actually permitted to use your admin? Are you simply allowing everyone, or do you have a whitelist defined somewhere that you are checking against?

3) Since this uses assets from `trestle-auth`, it should also be licensed under LGPLv3 (https://github.com/TrestleAdmin/trestle-auth#license).

4) With recent updates to `trestle-auth`, I am hoping to also support OmniAuth authentication. However this will remain a model-based solution.

5) If you are interested in allowing further customizations, it would be nice to be able to specify a color code and icon for a given provider. You may also want to add the `btn-block` class to the provider links [here](https://github.com/airhorns/trestle-omniauth/blob/master/app/views/trestle/omniauth/sessions/new.html.erb#L10).

airhorns commented:
> I've had a quick review, but unfortunately wasn't able to get it working (just testing on a new app with the developer provider and no existing OmniAuth integration).

Thanks for trying it!
Did some digging on this -- turns out the `:developer` strategy doesn't work with Rails CSRF protection. It uses a (IMO janky) POST right at the `/auth/developer/callback` URL without using Rails to build the form which triggers the issue. I could `skip_before_action :verify_authenticity_token` for that action but that feels like overkill and a little magical, so I just added a note to the readme about the strategy not working.

airhorns commented:
> How are you configuring OmniAuth to use the sessions#create action (which is mounted by default at /admin/auth/:provider/callback)?

I wasn't before, oops! I had forgotten Trestle had the `.path` config option. Updated to teach Omniauth about that and 🎩 'd locally.

> How are you authorizing that an OmniAuth-authenticated user is actually permitted to use your admin? Are you simply allowing everyone, or do you have a whitelist defined somewhere that you are checking against?

I am not. With Google which is what I've been using, you are able to manage some of this kind of thing on the side of the client id & secret such that it will only authenticate users from the organization that provisioned it. With other providers there are sometimes options to limit the scope as well. 

I added a note to the README about this, and I think that in the event someone needs more robust permission-ing / whitelisting, they should consider using `trestle-auth`. I think `trestle-omniauth` sits in a nice spot of being almost mindless to set up, whereas if you need something more "real" and are willing to invest there are better, more powerful steps up the ladder to take.

 > Since this uses assets from trestle-auth, it should also be licensed under LGPLv3 (https://github.com/TrestleAdmin/trestle-auth#license).

My bad! Truth be told this is the first time I've worked with non MIT code I think! I have relicensed `trestle-omniauth`.

> With recent updates to trestle-auth, I am hoping to also support OmniAuth authentication. However this will remain a model-based solution.

I think that makes sense and ultimately may be what more Trestle developers may end up going with, which is fine with me! If you think this thing is too simple or makes it too hard to figure out which auth solution to use for Trestle I am happy to let it be how it is and switch over to `trestle-auth` as well. I just want SSO :) 

> If you are interested in allowing further customizations, it would be nice to be able to specify a color code and icon for a given provider. You may also want to add the btn-block class to the provider links here.

Added the class, thanks!

spohlenz commented:
> I am not. With Google which is what I've been using, you are able to manage some of this kind of thing on the side of the client id & secret such that it will only authenticate users from the organization that provisioned it. With other providers there are sometimes options to limit the scope as well.

Ah, interesting! TIL. That in particular makes this a very useful alternative.

Thanks for your work on this.
